lib/poseidon/sync_producer.rb

```diff
@@ -57,6 +57,7 @@ def send_messages(messages)
         if !messages_to_send.pending_messages? || @max_send_retries == 0
           break
         else
+          Poseidon.logger.debug { "retrying sending #{messages_to_send.messages} messages"  }
```

Comment:
Yup exactly, if we set the log level to a higher level than debug, it won't call these blocks, which can be a performance win